src/main/java/Storage.java

```diff
@@ -0,0 +1,80 @@
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * Class to load and write tasks objects to the local drive.
+ * @author vanGoghhh
+ */
+
+public class Storage {
+
+    private String filepath;
+
+    /**
+     * Constructor for Storage.
+     * @param filepath specified filepath in the user's local drive.
+     */
+    public Storage(String filepath) {
+        this.filepath = filepath;
+    }
+
+    /**
+     * Loads data from the user's local files.
+     * @return the tasks in the users local files in an arraylist.
+     * @throws IOException
+     */
+    protected ArrayList<Task> loadData() throws IOException {
+            ArrayList<Task> taskList = new ArrayList<>();
+            Files.createDirectories(Paths.get("data"));
+            File dukeData = new File(this.filepath);
+            if (dukeData.exists()) {
+                Scanner dukeScanner = new Scanner(dukeData);
+                while (dukeScanner.hasNext()) {
+                    String txtLine = dukeScanner.nextLine();
+                    String[] txtLineArr = txtLine.split("//");
+                    Task task;
+                    if (txtLineArr[0].equals("T")) {
+                        task = new Todo(txtLineArr[2]);
+                        if (txtLineArr[1].equals("1")) {
+                            task.markAsDone();
+                        }
```

Comment:
Do you need to include this inside every if block, or can you have 1 of this outside the block?